### PR TITLE
Add missing '-race' flag to go build for race rpm

### PR DESCRIPTION
**Pull request by jordigilh**
Added `-race` to the `go build` command to build the race rpm with the race flag enabled.

### CODE REVIEW

File: flotta-agent.spec [hunk 1]
```diff
@@ -46,10 +46,10 @@ tar fx %{SOURCE0}
 
 %build
 cd flotta-agent-%{VERSION}
-export CGO_ENABLED=0
+export CGO_ENABLED=1
```
[the remark this answers is not in the document]
Comment (eloycoto) [reply flotta-agent.spec:50]: that's bad, all will have CGO_ENABLED, that need to be changed.

[the remark this answers is not in the document]
Comment (jordigilh) [reply flotta-agent.spec:50]: I think we could have the flag enabled just for the `-race`. Would have to define it twice though, but then we enable it for just the race package.
Btw... why would it be bad? I thought having it enabled created a statically linked binary.

[the remark this answers is not in the document]
Comment (jordigilh) [reply flotta-agent.spec]: Ok, enabled only for the `race` build.

File: flotta-agent.spec [hunk 2]
```diff
@@ -81,6 +83,9 @@ ln -sf %{_libexecdir}/yggdrasil/device-worker-race %{_libexecdir}/yggdrasil/devi
 systemctl enable --now nftables.service
 
 %changelog
+* Thu Jun 23 2022 Jordi Gil <jgil@redhat.com> 0.1.0-3
```
Comment (eloycoto) [inline flotta-agent.spec:86]: if you wantt to update the release, do the same on the makefile and on the top of the file

**Review by eloycoto — CHANGES_REQUESTED** [review]

**Review by eloycoto — APPROVED** [review]

**Review by masayag — APPROVED** [review]

**Pull request merged**